repos/spack_repo/builtin/packages/mvapich_plus/package.py

```diff
@@ -120,87 +121,47 @@ def libs(self):
 
         return find_libraries(libraries, root=self.prefix, shared=True, recursive=True)
 
-    @property
-    def process_manager_options(self):
-        spec = self.spec
-
-        other_pms = []
-        for x in ("hydra", "gforker", "remshell"):
-            if spec.satisfies(f"process_managers={x}"):
-                other_pms.append(x)
-
-        opts = []
-        if len(other_pms) > 0:
-            opts = ["--with-pm=%s" % ":".join(other_pms)]
-        if spec.satisfies("process_managers=slurm"):
-            opts = ["--with-pm=none", "--with-pmi=slurm"]
-        if spec.satisfies("process_managers=cray"):
-            opts = ["--with-pm=none", "--with-pmi=pmi2", "--with-pmi2=/opt/cray/pe/pmi/default"]
-        if "none" in spec.variants["process_managers"].value:
-            opts = ["--with-pm=none"]
-
-        return opts
+    def install(self, spec, prefix):
+        print(self.stage.source_path)
+        runfile = glob(join_path(self.stage.source_path, "*"))[0]
+        mvp_ver=str(spec.version)
+        gpu="nogpu"
+        gpu_ver=""
+        apu=''
+        if spec.satisfies("^cuda"):
+            gpu = "cuda"
+            gpu_ver=str(spec['cuda'].version)[:4]
+        elif spec.satisfies('+rocm'):
+            gpu = "rocm"
+            gpu_ver=spec['hip'].version
+            if spec.satisfies('+apu'):
+                apu='.mi300a'
+
+        netmod = "ucx" if spec.satisfies("netmod=ucx") else "ofi"
+        comp = str(spec.compiler)
+        comp = f"{comp[:comp.find('@')]}{comp[comp.find('@')+1:comp.find('/')]}"
+        el="el8"
```

Comment:
On a similar note, I would probably harden this a bit in case you build for more OSes in the future...